### ISSUE TITLE
Support customizable iterator records

### ISSUE THREAD

**Issue by mscappini** [jsgoupil/quickbooks-sync#7]
Currently, the `StepQueryWithIterator<T, Y>.ExecuteRequest` method simply sets the `QbIteratorRequest.MaxReturned` value to 100. We could check for an existing valid value and set one if there isn't.

``` cs
int maxVal;
bool isValid = int.TryParse(request.MaxReturned, out maxVal);

// Max 200 is arbitrary--maybe there shouldn't be a max.
if (!isValid || (isValid && (0 > maxVal || maxVal > 200)))
{
    // Not a valid max returned value. Set to 100 by default.
    request.MaxReturned = "100";
}
```

**Comment by jsgoupil** [comment]
:+1:

**Comment by mscappini** [comment]
Implemented 16749184c1110c473af098b9b65942566fff4600.
